### ISSUE TITLE
Windows installation via WinGet

### ISSUE THREAD

**Issue by spencerwooo** [ajeetdsouza/zoxide#79]
Anything planned for Windows? Preferably: installation via WinGet/Scoop, publishing Windows binaries alongside Linux/macOS on GitHub release, etc.

**Comment by ajeetdsouza** [comment]
Absolutely! It may take a while since I'm not familiar with packaging on Windows - if you have experience, I'd appreciate any help!

**Comment by spencerwooo** [comment]
I believe for `zoxide` to take advantage of installing via Scoop/WinGet, we must first setup build scripts for Windows on GitHub release. I'll make a PR for the GitHub Action CI first, and if all goes well, I'll poke around and try adding `zoxide` to Scoop/WinGet's repo next.

(Sorry for the late response...I've been extra busy.)

**Comment by ajeetdsouza** [comment]
Regarding CI, I think it might be better to switch over to the [trust](https://github.com/japaric/trust) project. It has testing, releases on multiple architectures, and a shell-based install script.

I've already done some work on this, I'll ping you here once it's done!

**Comment by spencerwooo** [comment]
Ok, so we won't be using GitHub Actions anymore?

**Comment by ajeetdsouza** [comment]
Probably not - if trust takes care of deployment and installation of binaries, I think we should shift to what they're using.

**Comment by ajeetdsouza** [comment]
@spencerwooo it appears trust's install script relies on `rustc` unless `target` is explicitly specified. IMHO, this defeats its purpose.

I think we can go ahead with GitHub Actions for now. Feel free to send in a PR once you have the time!

**Comment by ajeetdsouza** [comment]
winget support is currently waiting on https://github.com/microsoft/winget-cli/issues/182.

**Comment by spencerwooo** [comment]
I just [created a pull request](https://github.com/ScoopInstaller/Main/pull/1288) over at scoop's main bucket for `zoxide` version 0.4.3. If merged, we'll be able to install `zoxide` with a simple command on Windows:

```powershell
scoop install zoxide
```

I'll update the documentations here once it's merged upstream (if that's ok with you).

**Comment by ajeetdsouza** [comment]
Looks great. Thank you!

**Comment by mhetrick** [comment]
I just tried installing this via `scoop`, but I'm unfortunately hitting a hash check failure. Is this something I should report to the Scoop team or is it suitable reporting it here?

![image](https://user-images.githubusercontent.com/416077/122088215-26ec2f00-cdcb-11eb-8874-09c68317df49.png)

**Comment by ditsuke** [comment]
@spencerwooo can you please close this issue?

**Comment by ajeetdsouza** [comment]
@ditsuke I left the issue open because zoxide hasn't been added to WinGet yet. Currently, it's blocked on https://github.com/microsoft/winget-cli/issues/140, but once that happens, I'll close the issue.

**Comment by ajeetdsouza** [comment]
Support for portable exes has been merged into Winget. Tracking the package request here: https://github.com/microsoft/winget-pkgs/issues/80892

**Comment by sitiom** [comment]
It's coming!
- https://github.com/microsoft/winget-pkgs/pull/94715

Once the PR has been merged, [Winget Releaser](https://github.com/vedantmgoyal2009/winget-releaser) must be set up in this repo to push automatic manifest updates. @ajeetdsouza Are you good with that?

**Comment by ajeetdsouza** [comment]
Yeah, that sounds great!

**Comment by kingyue737** [comment]
zoxide now available on winget

![image](https://user-images.githubusercontent.com/40021217/215648053-ccf43535-1ce6-493c-9ae9-4647b2964a70.png)


Installed successfully. Thanks @sitiom !

**Comment by sitiom** [comment]
For the Winget Releaser workflow, https://github.com/vedantmgoyal2009/winget-releaser/issues/82 would have to be resolved first.

**Comment by ajeetdsouza** [comment]
Thanks @sitiom, really appreciate your work on this! I'll close this issue for now, but I'd be very happy to merge a later PR for auto-updating the package on winget.
